### DISCUSSION TITLE
Difference between ENCODE CLIP and TAG OBJECTS

### DISCUSSION THREAD

**Discussion by jagjordi** [immich-app/immich#2071] [Q&A]
Hello
What is the difference between the ENCODE CLIP and the TAG OBJECTS jobs?
It looks to me that they both are for generating the "metadata" used for searching

**Comment by bo0tzz** [comment]
The object tagging uses a machine learning model to identify objects in the photo. This results in labels that we can attach to the metadata, such as "chair", "boat", "cat", etc. We can then return matching photos if you search for exactly that label.

The CLIP encoding is a bit more complicated. It feeds the image into a machine learning model and extracts some internal state of the model afterwards. This is a series of numbers that you can think of as coordinates (but in many dimensions). Later, when searching, we do the same thing to your search query - feed it into the model, and turn it into coordinates. We can then find any photos whose coordinates are close to those of the search query. This lets you search more loosely based on what is actually in the image, like "a cat on a chair" or "a boat in a storm".

**Reply by klejejs** [reply]
@bo0tzz I noticed that when I clicked on "Generate Thumbnails -ALL", loading icons appeared also for "Tag Objects" and "Generate Clip". Do I have to click on each entry separately to generate all data, or does one click run machine model and generate everything?

**Reply by bo0tzz** [reply]
After generating a thumbnail, that job automatically queues up tasks for generating the ML data as well.
